enginetest/enginetests.go

```diff
@@ -3855,229 +2869,20 @@ func TestDisallowedCheckConstraints(t *testing.T, harness Harness) {
-	AssertErr(t, e, harness, "ALTER TABLE t1 ADD CONSTRAINT chk2 CHECK (is_free_lock('abc') is null)", sql.ErrInvalidConstraintFunctionNotSupported)
-	AssertErr(t, e, harness, "ALTER TABLE t1 ADD CONSTRAINT chk2 CHECK (is_used_lock('abc') is null)", sql.ErrInvalidConstraintFunctionNotSupported)
-
-	// subqueries
-	AssertErr(t, e, harness, "ALTER TABLE t1 ADD CONSTRAINT chk2 CHECK ((select count(*) from t1) = 0)", sql.ErrInvalidConstraintSubqueryNotSupported)
-
 	// TODO: need checks for stored procedures, also not allowed
-
-	// Some spot checks on create table forms of the above
-	AssertErr(t, e, harness, `
-CREATE TABLE t3 (
-	a int primary key CONSTRAINT chk2 CHECK (current_user = "root@")
-)
-`, sql.ErrInvalidConstraintFunctionNotSupported)
-
-	AssertErr(t, e, harness, `
-CREATE TABLE t3 (
-	a int primary key,
-	CHECK (current_user = "root@")
-)
-`, sql.ErrInvalidConstraintFunctionNotSupported)
-
-	AssertErr(t, e, harness, `
-CREATE TABLE t3 (
-	a int primary key CONSTRAINT chk2 CHECK (a = (select count(*) from t1))
-)
-`, sql.ErrInvalidConstraintSubqueryNotSupported)
-
-	AssertErr(t, e, harness, `
-CREATE TABLE t3 (
-	a int primary key,
-	CHECK (a = (select count(*) from t1))
-)
-`, sql.ErrInvalidConstraintSubqueryNotSupported)
-}
-
-// todo(max): rewrite with []ScriptTest
-func TestDropCheckConstraints(t *testing.T, harness Harness) {
-	require := require.New(t)
-
-	harness.Setup(setup.MydbData)
-	e := mustNewEngine(t, harness)
-	defer e.Close()
-	ctx := NewContext(harness)
-
-	RunQuery(t, e, harness, "CREATE TABLE t1 (a INTEGER PRIMARY KEY, b INTEGER, c integer)")
-	RunQuery(t, e, harness, "ALTER TABLE t1 ADD CONSTRAINT chk1 CHECK (a > 0)")
-	RunQuery(t, e, harness, "ALTER TABLE t1 ADD CONSTRAINT chk2 CHECK (b > 0) NOT ENFORCED")
-	RunQuery(t, e, harness, "ALTER TABLE t1 ADD CONSTRAINT chk3 CHECK (c > 0)")
-	RunQuery(t, e, harness, "ALTER TABLE t1 DROP CONSTRAINT chk2")
-	RunQuery(t, e, harness, "ALTER TABLE t1 DROP CHECK chk1")
-
-	db, err := e.EngineAnalyzer().Catalog.Database(ctx, "mydb")
-	require.NoError(err)
-
-	table, ok, err := db.GetTableInsensitive(ctx, "t1")
-	require.NoError(err)
-	require.True(ok)
-
-	cht, ok := table.(sql.CheckTable)
-	require.True(ok)
-
-	checks, err := cht.GetChecks(NewContext(harness))
-	require.NoError(err)
-
-	expected := []sql.CheckDefinition{
-		{
-			Name:            "chk3",
-			CheckExpression: "(c > 0)",
-			Enforced:        true,
-		},
+	for _, tt := range queries.DisallowedCheckConstraintsScripts {
+		TestScriptWithEngine(t, e, harness, tt)
 	}
-
-	assert.Equal(t, expected, checks)
-
-	RunQuery(t, e, harness, "ALTER TABLE t1 DROP CHECK chk3")
-
-	// Some faulty drop statements
-	AssertErr(t, e, harness, "ALTER TABLE t2 DROP CONSTRAINT chk2", sql.ErrTableNotFound)
-	AssertErr(t, e, harness, "ALTER TABLE t1 DROP CONSTRAINT dne", sql.ErrUnknownConstraint)
 }
 
-func TestDropConstraints(t *testing.T, harness Harness) {
```

Comment:
Removed `TestDropConstraints` as the test coverage is included in `TestDropCheckConstraints` and `TestDropForeignKeys`